include/aspect/material_model/rheology/composite_visco_plastic.h

```diff
@@ -108,14 +111,28 @@ namespace aspect
             * each phase and then averaged for each compositional field.
             */
           std::pair<double, double>
-          compute_strain_rate_and_derivative (const double creep_stress,
-                                              const double pressure,
-                                              const double temperature,
-                                              const unsigned int composition,
-                                              const DiffusionCreepParameters diffusion_creep_parameters,
-                                              const DislocationCreepParameters dislocation_creep_parameters,
-                                              const PeierlsCreepParameters peierls_creep_parameters,
-                                              const DruckerPragerParameters drucker_prager_parameters) const;
+          compute_creep_strain_rate_and_derivative (const double creep_stress,
+                                                    const double pressure,
+                                                    const double temperature,
+                                                    const unsigned int composition,
+                                                    const DiffusionCreepParameters diffusion_creep_parameters,
+                                                    const DislocationCreepParameters dislocation_creep_parameters,
+                                                    const PeierlsCreepParameters peierls_creep_parameters,
+                                                    const DruckerPragerParameters drucker_prager_parameters) const;
+
+          /**
```

Comment:
I'm a bit confused about the need for this function as `elasticity->strain_rates_from_stored_elastic_stresses` is the main operation done within it along with a check to make sure elasticity is used. Can you provide a brief explanation above about why a direct call the function in the elasticity module would not work?

Reply:
I think you're right. It might not even be needed at all.